src/interface/metadata.hpp

```diff
@@ -573,7 +573,10 @@ class Metadata {
   const std::string &getAssociated() const { return associated_; }
 
   void SetFluxName(const std::string &name) { flux_var_ = name; }
-  const std::string &GetFluxName() const { return flux_var_; }
+  const std::string &GetFluxName() const {
+    PARTHENON_DEBUG_REQUIRE(IsSet(WithFluxes), "Variable has a flux.");
```

Comment:
This is probably good to have for now, but in principle a variable could be declared without `Metadata::WithFluxes`, another variable could be explicitly declared `Metadata::Flux`, and then the two could be associated. Maybe the thing to do though is to set the `WithFluxes` flag on the first variable when a flux is associated with it though.

Reply:
in the long term, could change this to `flux_var_.size() > 0`. But I'd like to leave it for now.